### ISSUE TITLE
[Bug Report] ttnn.maximum fails when one of the arguments is -np.inf

### ISSUE THREAD

**Issue by umalesTT** [tenstorrent/tt-metal#18556]
**Describe the bug**
ttnn.maximum fails when one of the arguments is negative infinity (-np.inf). Output is 
``` ttnn.Tensor([    -nan], shape=Shape([1]), dtype=DataType::FLOAT32, layout=Layout::TILE) ```

**To Reproduce**
1. Run 
```
import ttnn
import torch
import numpy as np
with ttnn.manage_device(device_id=0) as device:
    a = ttnn.from_torch(torch.tensor([1.0]), layout = ttnn.TILE_LAYOUT, device=device)
    b = ttnn.from_torch(torch.tensor([-np.inf]), layout = ttnn.TILE_LAYOUT, device=device)
    res = ttnn.maximum(b, a)
    print(res)
```

**Expected behavior**
Expected output is ``` ttnn.Tensor([1.0], shape=Shape([1]), dtype=DataType::FLOAT32, layout=Layout::TILE) ```

**Additional context**
This is the reason why 1D softmax is not working in JAX; within implementation of jax.nn.softmax -jnp.inf appears when calculating max across tensor (which afterwards results in comparison of scalar and -jnp.inf).

**Please complete the following environment information:**
Version of software (eg. commit) https://github.com/tenstorrent/tt-metal/commit/9b3d855773

**Comment by VirdhatchaniKN** [comment]
Hi @cmaryanTT @rtawfik01 @rdjogoTT 
Regarding maximum issue, the op passes for bfloat16 data type but fails for float32 data type. Below is our analysis

- For maximum implementation ( File: `ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_composite_op.cpp` ), below attached is the result comparison. Based on this report ([nan_inf_behaviour_WH_BH.pdf](https://github.com/user-attachments/files/19428420/nan_inf_behaviour_WH_BH.pdf) - #19051) , 
  - BFLOAT16: 0 * -inf = 0, this behaviour enables us to use multiply in composite op implementations for bfloat16 dtype.
  - FLOAT32: 0 * -inf = -nan, this behaviour (although it is correct as per torch.mul) it affects composite op implementations for float32 dtype
 
<img width="285" alt="Image" src="https://github.com/user-attachments/assets/ba6e55f1-8067-4cfb-b53f-4e99aeae2d59" />

- Using typecast just to handle this edge-case may not be the right approach as it will affect precision and accuracy of the remaining values
- Tried different logic for maximum but the multiplication becomes a blocker there as well.
- Inputs required on if `0 * -inf = 0` is possible
  - The difference between two data types is the behaviour of multiplication with nan/inf and 0.
  - Is it possible for the `mul_binary_tile()` api to have two modes of operation where it can give the result `0 * -inf = -nan` or `0 * -inf = 0` based on the selected mode?
  - This way, Fp32 dtype multiplication will have two modes. One is the current implementation, second is to result in 0 which will enable composite op implementations involving edge cases.

**Comment by VirdhatchaniKN** [comment]
Update : 

- Completed report with the difference in edge case for nan inf cases between Bfloat16 and Float32 for add, sub, multiply, div
  - PR for testing : https://github.com/tenstorrent/tt-metal/pull/19051
  - Survey Report with summary of behaviour : [nan_inf_behaviour_WH_BH.pdf](https://github.com/user-attachments/files/19651929/nan_inf_behaviour_WH_BH.pdf)
  - Problem description : [Maximum op issue for edge case.pptx](https://github.com/user-attachments/files/19651960/Maximum.op.issue.for.edge.case.pptx)
- Based on discussion, the next steps would be to add SFPU implementation for min, max instead of composite implementation, which would improve performance and also handle inf case as mentioned in this issue
  - #19974 : Implement Tensor-scalar version of max, min 
    - Completed and merged to main
    - **Note** : Nan behaviour for Tensor-scalar version of min,max : [Link](https://github.com/tenstorrent/tt-metal/issues/19974#issuecomment-2783668195)
  - #20241 : Implement Tensor-Tensor version of Min, Max
    - In Progress

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/pull/20390 - Merged
Issue fixed, Hence closing issue
